### PR TITLE
Document HookContext with param and return types

### PR DESCRIPTION
**Pull request by ekohl**
What made me do this was that the examples don't render well (see https://www.rubydoc.info/gems/kafo/5.1.0/Kafo/HookContext). Adding data types makes the documentation a bit more useful.

### CODE REVIEW

File: lib/kafo/hook_context.rb
Comment (wbclark) [inline lib/kafo/hook_context.rb:10]:
```suggestion
    # Hooks in `pre_migrations` and `boot` stages won't print any message because the logger configuration
    # depends on the application configuration, which is not yet available.
```

Might as well improve this a bit while we are here. I'm unsure though if 'available' is the right word here, maybe 'complete' is better.

**Comment by wbclark** [comment]
Not really a blocker, but for my benefit, do you have an example (from another project?) of how these docs are rendered with param and return types?

**Review by wbclark — APPROVED** [review]
I'll approve this and leave it up to you if you want to address the above nit before merging.

**Comment by ekohl** [comment]
> Not really a blocker, but for my benefit, do you have an example (from another project?) of how these docs are rendered with param and return types?

[yard](https://yardoc.org/guides/index.html) is used. There is a **lot** more we can do. For example, `@param` also accepts an actual description of each parameter. I thought the names were self documenting enough for now.

File: lib/kafo/hook_context.rb
Comment (ekohl) [reply lib/kafo/hook_context.rb:10]: Initialized might be a better word, but really it shouldn't be used in those stages. I've actually been thinking about how we can actually remove them from the context. Perhaps it should raise an exception.

Comment (ehelms) [reply lib/kafo/hook_context.rb:10]: This is a bit confusing as to what it means by won't print any messages. Did this mean in the log file? On the console? When using the progress bar? I know this is true for the progress bar output, but its not true for the log file or console based output that comes with verbose.

Comment (ekohl) [reply lib/kafo/hook_context.rb:10]: Given this is only moving the existing documentation I'm going to merge it now. There is plenty of room for more improvement.

**Pull request merged**
